echo-scheduler/src/main/groovy/com/netflix/spinnaker/echo/scheduler/actions/pipeline/impl/PipelineConfigsPollingAgent.groovy

```diff
@@ -173,48 +170,46 @@ class PipelineConfigsPollingAgent extends AbstractPollingAgent {
     }
   }
 
-  void registerNewTriggers(List<Pipeline> pipelines, List<ActionInstance> actionInstances) {
+  void registerNewTriggers(TriggerRepository triggerRepo, List<ActionInstance> actionInstances) {
     try {
-      List<String> actionIds = actionInstances.collect { it.id }
-
-      /**
-       * Find all pipelines that don't have a trigger that is not associated (by id) with any actionInstance in the
-       * list of actionInstances. This indicates that these are new triggers
-       */
-      List<Pipeline> pipelinesWithNewTriggers = pipelines.findAll {
-        def cronTriggers = it.triggers.findAll { it.type == Trigger.Type.CRON.toString() }
-        cronTriggers.any { !actionIds.contains(it.id) && !actionIds.find { actionId -> actionId.endsWith(":$it.id") } }
+      // find all the triggers that don't have a corresponding action instance (by id)
+      // this indicates that these are new triggers
+      actionInstances.each { actionInstance ->
+        triggerRepo.remove(actionInstance.id)
       }
-      pipelinesWithNewTriggers.each { pipeline ->
-        /**
-         * Extract all the triggers of type 'cron' from a pipeline. Ideally, there would be just one,
-         * but pipelines 'could' have multiple cron triggers
-         */
-        List<Trigger> triggers = pipeline.triggers.findAll {
-          it.type == Trigger.Type.CRON.toString() && it.enabled
+
+      Collection<Trigger> newTriggers = triggerRepo.triggers()
+      newTriggers.each { trigger ->
+        if (!Trigger.Type.CRON.toString().equalsIgnoreCase(trigger.type) || !trigger.enabled) {
+//          log.debug("Skipping disabled or non-cron trigger ${trigger}")
+          return
         }
-        if (triggers) {
-          log.info("Found '${triggers.size()}' new cron trigger(s) for ${pipeline}...")
+
+        Pipeline pipeline = trigger.parent
+        if (pipeline.disabled) {
```

Comment:
we used to miss this case, we would scheduled triggers for disabled pipelines

Reply:
Good catch!